app/javascript/lib/validation/validate-file-content.js

```diff
@@ -551,17 +693,18 @@ export async function parseClusterFile(chunker, mimeType) {
   return { issues, delimiter, numColumns: headers[0].length }
 }
 
-/** reads in the specified number of header lines, sniffs the delimiter, and returns the
- * lines parsed by the sniffed delimiter
+/**
+ * reads in a two lines to be used as header lines, sniffs the delimiter,
+ * and returns the lines parsed by the sniffed delimiter
  */
-export async function getParsedHeaderLines(chunker, mimeType, numHeaderLines=2) {
+export async function getParsedHeaderLines(chunker, mimeType) {
```

Comment:
BLOCKER: you need to only read one header line for dense matrices (the list of cell names).  IF you read the first two lines as headers, then the second line of the file (which is the first gene), won't get passed into the regular validation routines

Reply:
This `numHeaderLines` has to be 2 since the logic to determine the delimiter done within this `getParsedHeaderLines()` function requires 2 lines. Removing this argument from the header was done to remove the confusion that you could specify only 1 header line here, as that will break the delimiter sniffer.

Since Dense Matrices are the only ones using this `getParsedHeaderLines()` function but only need 1 header line I think this is work to figure out a better handling of a single header line would make most sense to be addressed in the "Allow R-fomatted Dense Matrices" ticket since that is what is breaking allowing an R-formatted matrix.